streams/src/main/java/org/apache/kafka/streams/kstream/Consumed.java

```diff
@@ -96,13 +112,21 @@ protected Consumed(final Consumed<K, V> consumed) {
      *
      * @return a new instance of {@link Consumed}
      */
+    @Deprecated
     public static <K, V> Consumed<K, V> with(final Serde<K> keySerde,
                                              final Serde<V> valueSerde,
                                              final TimestampExtractor timestampExtractor,
                                              final Topology.AutoOffsetReset resetPolicy) {
         return new Consumed<>(keySerde, valueSerde, timestampExtractor, resetPolicy, null);
     }
 
+    public static <K, V> Consumed<K, V> with(final Serde<K> keySerde,
+                                             final Serde<V> valueSerde,
+                                             final TimestampExtractor timestampExtractor,
+                                             final AutoOffsetReset resetPolicy) {
+        return new Consumed<>(keySerde, valueSerde, timestampExtractor, resetPolicy, null);
```

Comment:
We can pass `legacyResetPolicy` as `null`.

Reply:
Updated the Consumed methods to the best of my understanding.. I am not feeling confident that I made the change you were after. We are adding a helper function to be able to pass the legacy version as the new one to satisfy the refactored constructor, right? Was it appropriate to pass it twice for the legacy reset policy methods, and once for the methods that use the new one?

Please double check me on this and I will change it if needed, I think I've committed what you were after here.